clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -675,6 +688,7 @@ protected Map<String, Object> postProcessParsedConfig(final Map<String, Object>
         maybeOverrideClientId(refinedConfigs);
         maybeOverrideEnableAutoCommit(refinedConfigs);
         checkGroupRemoteAssignor();
+        checkUnsupportConfigsWithConsumerProtocol();
```

Comment:
We can call the more generalized `checkUnsupportedConfigs()` method directly:

```suggestion
        checkUnsupportedConfigs(GroupProtocol.CLASSIC, CLASSIC_PROTOCOL_UNSUPPORTED_CONFIGS);
        checkUnsupportedConfigs(GroupProtocol.CONSUMER, CONSUMER_PROTOCOL_UNSUPPORTED_CONFIGS);
```